components/Chat/ChatSettings/SettingsMenu.vue

```diff
@@ -218,13 +255,19 @@
         getUseIgnore      : Chat.$states.useIgnore,
         getTrollTts       : Chat.$states.trollTts,
         getTtsRate        : Chat.$states.ttsRate,
+        getTtsReadUsername: Chat.$states.ttsReadUsername,
         getTtsTimeout     : Chat.$states.ttsTimeout,
         getTtsVolume      : Chat.$states.ttsVolume,
         getTtsVoice       : Chat.$states.ttsVoice,
         getNotify         : Chat.$states.notify,
         getAutocomplete   : Chat.$states.autocomplete,
+        getRecieveMentionsInLocal : Chat.$states.recieveMentionsInLocal,
       }),
 
+      ttsVoicesExist: {
+        get () { return this.ttsVoices.length }
```

Comment:
`length` isn't a boolean, so it's technically not type-safe to use it as one (or at least I'd imagine, when we change to typescript). `!!this.ttsVoices.length`? I've no experience with these types of *possible* errors.

Reply:
For a typescript conversion I've always ended up using shims for stuff like that. something like
`return this.ttsVoices && this.ttsVoices.length` where ttsVoices is `array?`, in the event that it's undefined or null, it will short circuit the length property check and therefore won't throw any errors. In reality this just silently fails and gets logged via sentry so eventually it'll come up.